app/components/courses/search_result_component.rb

```diff
@@ -12,10 +12,11 @@ def has_sites?
       @has_sites
     end
 
-    def initialize(course:, filtered_by_location: false, has_sites: false)
+    def initialize(course:, results_view:)
       @course = course.decorate
-      @filtered_by_location = filtered_by_location
-      @has_sites = has_sites
+      @filtered_by_location = results_view.location_filter?
+      @has_sites = results_view.has_sites?(course)
```

Comment:
I don't think `@has_sites` needs to be declared as an instance var?

Perhaps just define `#has_sites` as follows 

```
 def has_sites?
     @results_view.has_sites?(@course)
  end
```

Reply:
Same deal for @filtered_by_location